### PR TITLE
LPS-198957 Allow send notification to organization and site roles

### PR DESCRIPTION
**Pull request by seltong**
Related issue: [LPS-198957](https://liferay.atlassian.net/browse/LPS-198957)

### CODE REVIEW

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/context/NotificationContext.java [hunk 1]
```diff
@@ -47,6 +47,10 @@ public List<Long> getFileEntryIds() {
 		return _fileEntryIds;
 	}
 
+	public Long getGroupId() {
```
Comment (murilostodolni) [inline modules/apps/notification/notification-api/src/main/java/com/liferay/notification/context/NotificationContext.java:50]: Could you change to `long`, please?

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/context/NotificationContext.java [hunk 2]
```diff
@@ -148,6 +156,7 @@ public void setUserId(long userId) {
 	private long _classPK;
 	private String _externalReferenceCode;
 	private List<Long> _fileEntryIds;
+	private Long _groupId;
```
Comment (murilostodolni) [inline modules/apps/notification/notification-api/src/main/java/com/liferay/notification/context/NotificationContext.java:159]: Same here.

**Review by murilostodolni — CHANGES_REQUESTED** [review]
Hey @seltong 

I finished my review. Let me know if you have questions.

**Review by murilostodolni — APPROVED** [review]
LGTM 🎉

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/users/provider/RoleUsersProvider.java [hunk 3]
```diff
@@ -59,6 +63,18 @@ public List<User> provide(NotificationContext notificationContext)
...
 				notificationRecipientSetting.getValue());
 
+			if ((role.getType() == RoleConstants.TYPE_ORGANIZATION) ||
+				(role.getType() == RoleConstants.TYPE_SITE)) {
+
+				userIds.addAll(
+					ListUtil.toList(
+						_userGroupRoleLocalService.getUserGroupRolesByGroup(
+							notificationContext.getGroupId()),
+						UserGroupRoleModel::getUserId));
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/users/provider/RoleUsersProvider.java:73]: @seltong could you please check if we can use the `_userLocalService.getGroupUserIds`?

Comment (seltong) [reply modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/users/provider/RoleUsersProvider.java:73]: We can use the `_userLocalService.getGroupUserIds` to get the userIds that are linked to site roles, but we cannot use it to get userIds linked to organization roles.

I tryied to use the `_userLocalService.getOrganizationUserIds()`, but in this case we need a organizationId.

I'm inclined to go with the approach current approach.

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
@seltong please fix the [SemanticVersioningTest](https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/406049/testReport/com.liferay.semantic.versioning/SemanticVersioningTest) failure.

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:reevaluate:1387401_9379

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:forward

**Pull request closed without merging**